### PR TITLE
Lensauth

### PR DESCRIPTION
**Pull request by XxSNiPxX**
Added Lens Protocol Login and Register for their Test net.

Added Protected routes for the dashboard.

Error Code ERESOLVE: There's a dependency conflict between react-scripts@5.0.1 and the installed typescript@5.0.4. react-scripts requires a TypeScript version compatible with ^3.2.1 || ^4, but typescript@5.0.4 exceeds this range.

Solutions include downgrading TypeScript, using --legacy-peer-deps or --force during installation, or updating dependencies for compatibility.

### CODE REVIEW

**Comment by geolffreym** [comment]
@Jadapema @XxSNiPxX  this issue is related to create-react-app? Can we improve the development stack migrating to vite?  
https://github.com/WatchItDev/watchit-app/issues/216

**Comment by XxSNiPxX** [comment]
> @Jadapema @XxSNiPxX this issue (error Code ERESOLVE) is related to create-react-app? Can we improve the development stack migrating to vite? #216

It is yeah, but it works and should not matter. Just force install the libraries. The dependency issue is with some web3  libraries that are not currently used.

**Comment by Jadapema** [comment]
Hi @XxSNiPxX,

Thanks for the Lens Auth integration you've done. I wanted to share some ideas and suggestions to improve the implementation:

**SessionContext:** It would be ideal to handle everything related to login, logout, etc., from the sessionContext you expose. This way, we could allow users to explore the application and only request login when they try to interact with certain functionalities, possibly through a modal. If we adopt this approach, we might also need to remove or adjust the current login routes to ensure a smooth user experience. I’ll take care of the design tasks related to the login dynamics to move forward on this issue and have better clarity on how it should work.

**ESLint Warnings:** It would be nice to remove some ESLint warnings that are appearing, I think they are caused by unused imports. We could also improve the vertical format by removing unnecessary whitespace.

**Import Grouping:** We could group imports by context, for example, separate imports from external libraries from local ones, to improve code clarity.

**npmrc Temporary Fix**: I added a .npmrc so that it can be installed normally with npm i, but it's a temporary fix. Eventually, we should resolve the dependency conflict more permanently.

Thanks again for your work. I'm excited to move forward with these tweaks and improvements.

File: src/sections/login/LoginOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            // Add your custom logic for account creation here
+            console.log('Register with username:', username);
+            const client = new LensClient({  environment: development, // wont't work with `production`
+
+            });
+            const result = await client.wallet.createProfileWithHandle({  // e.g. 'alice' which will be '@lens/alice' in full-handle notation
+            handle: username,  to: wallet,});
+            console.log(result)
+            if (!isRelaySuccess(result)) {  console.error(`Something went wrong`, result); }
+              window.location.reload();
```
Comment (geolffreym) [inline src/sections/login/LoginOptions.tsx:63]: just wondering if we should have a component to display the issue here?

File: src/config-global.ts [hunk 2]
```diff
@@ -33,3 +33,4 @@ export const MAPBOX_API = process.env.REACT_APP_MAPBOX_API;
 
 // ROOT PATH AFTER LOGIN SUCCESSFUL
 export const PATH_AFTER_LOGIN = paths.dashboard.root; // as '/dashboard'
+export const PATH_BEFORE_LOGIN = '/login'; // as '/dashboard'
```
Comment (geolffreym) [inline src/config-global.ts:36]: paths is a hub for paths? it worth to keep all the routes in there?

**Review by geolffreym — APPROVED** [review]
LGTM after check the comments..

**Review by Jadapema — CHANGES_REQUESTED** [review]
Hi @XxSNiPxX,

Thank you for the Lens Auth integration you've done. I’ve left some comments in the code with a few ideas and suggestions to improve the implementation.

One suggestion is regarding the login routes. What do you think about allowing users to log in from any part of the app, without blocking them from viewing catalogs and movie information? Instead, we could prompt them to log in only when they try to interact with certain functionalities. I left a comment in the context about this, as it would require us to eliminate the login routes and handle everything in the context. We could then create a new task to integrate a login modal.

Additionally, it would be great to address some ESLint warnings that are appearing, which I believe are caused by unused imports.

Thanks again for your hard work. I'm excited to move forward with these tweaks and improvements.

**Pull request closed without merging**